cfpb/consumerfinance.gov
# TDP: Use native fetch

Per https://github.com/cfpb/consumerfinance.gov/pull/7349 we can migrate to native `fetch` and replace the wrapper TDP uses.

## Changes

- Replace wrapped xhr with native `fetch` implementation for TDP search.


## How to test this PR

1. `yarn build`
2. Visit http://localhost:8000/consumer-tools/educator-tools/youth-financial-education/teach/activities/ and test out some searches (e.g. `test` = good; and `test2` = no results).
3. Test out the search filters and see that the fetch request aborts.
